### PR TITLE
Moved everything that is generated to the generated folder

### PR DESCRIPTION
**Pull request by RazvanN7**
After the #6562 PR there still where some source files that where generated by ddmd/backend/optagen.c and ddmd/idgen.d. Those files where generated in ddmd or ddmd/backend while they rightfully belong to the generated folder. This PR fixes this.

### CODE REVIEW

**Comment by wilzbach** [comment]
Would you be so kind to update the [`.gitignore`](https://github.com/dlang/dmd/blob/master/.gitignore) as well? (e.g. adding `/generated` and removing the no longer possible excludes)

**Comment by RazvanN7** [comment]
@wilzbach sure, I just hope I get it right.

**Comment by RazvanN7** [comment]
@wilzbach Done.

File: .gitignore [hunk 1]
```diff
@@ -1,34 +1,13 @@
...
+test/trace.log
+generated/
 *.obj
-src/idgen.exe
 src/dmd.exe
-src/dmd_frontend.exe
-src/optabgen.exe
-src/verstr.h
 *.map
-.DS_Store
```
Comment (wilzbach) [inline .gitignore]: I think this one should stay on this list: https://en.wikipedia.org/wiki/.DS_Store

File: .gitignore [hunk 2]
```diff
@@ -1,34 +1,13 @@
...
-trace.def
-trace.log
```
Comment (wilzbach) [inline .gitignore]: What was your motivation to move these to `test/trace.{def,log}`?

File: src/posix.mak [hunk 3]
```diff
@@ -348,24 +348,24 @@ $G/glue.a: $(G_GLUE_OBJS)
 $G/backend.a: $(G_OBJS)
 	$(AR) rcs $@ $(G_OBJS)
 
-$G/dmd_frontend: $(FRONT_SRCS) $D/gluelayer.d $(ROOT_SRCS) $G/newdelete.o $(STRING_IMPORT_FILES) $(HOST_DMD_PATH)
+$G/dmd_frontend: $(FRONT_SRCS) $D/gluelayer.d $(ROOT_SRCS) $G/newdelete.o $(STRING_IMPORT_FILES) $(HOST_DMD_PATH) $G/id.d
```
Comment (wilzbach) [inline src/posix.mak]: Hmm why don't you add `$G/id.d` to `FRONT_SRCS`?
As you can see below it's quite easy to forgot and logically it is a frontend source file

**Review by wilzbach — COMMENTED** [review]
I think you are on a very good way!

In summary:
- mind the `.gitignore`
- mind the merge conflict
- add `$G/id.d` to the `$(FRONT_SRCS)` to avoid forgetting it accidentally

**Comment by RazvanN7** [comment]
@wilzbach @andralex @WalterBright  I am having second thoughts regarding the location of the generated source file id.d . Although it would be nice to have everything that is generated in the generated/ folder, I strongly suggest we keep id.d in ddmd/. I, honestly, don't know what is the reason behind generating id.d, but this leads to some inconsistencies:

- the module is called ddmd.id although it is located in generated/os/build/model (I suppose this can be changed, but wither way we can't call it generated..id;)

- the lexer library depends on this file, so it would be recommended to keep it in the same directory as the other files from which the library is generated

- #6581 complicates things for generating ddoc (see CyberShadow tester fail) when id.d is in the generated/ folder

Although there are workarounds for all the above mentioned, I think that it would be cleaner and easier to just either generate id.d in ddmd/ or keep it as a non-generated file (not sure what this implies).

**Comment by andralex** [comment]
Yah, it's okay to leave id.d there. Thanks!

**Comment by andralex** [comment]
cool! rock'n'roll!

**Review by andralex — APPROVED** [review]

**Pull request merged**